tests/components/dyson/test_vacuum.py

```diff
@@ -1,189 +1,145 @@
-        assert component.name == "Device_Vacuum"
-        assert component.is_on
-        assert component.status == "Cleaning"
-        assert component2.status == "Unknown"
-        assert component.battery_level == 85
-        assert component.fan_speed == "Quiet"
-        assert component.fan_speed_list == ["Quiet", "Max"]
-        assert component.device_state_attributes["position"] == "(0, 0)"
-        assert component.available
-        assert component.supported_features == 255
-        assert component.battery_icon == "mdi:battery-80"
-        assert component3.battery_icon == "mdi:battery-charging-40"
-
-    def test_turn_on(self):
-        """Test turn on vacuum."""
-        device1 = _get_vacuum_device_charging()
-        component1 = Dyson360EyeDevice(device1)
-        component1.turn_on()
-        assert device1.start.called
-
-        device2 = _get_vacuum_device_pause()
-        component2 = Dyson360EyeDevice(device2)
-        component2.turn_on()
-        assert device2.resume.called
-
-    def test_turn_off(self):
-        """Test turn off vacuum."""
-        device1 = _get_vacuum_device_cleaning()
-        component1 = Dyson360EyeDevice(device1)
-        component1.turn_off()
-        assert device1.pause.called
-
-    def test_stop(self):
-        """Test stop vacuum."""
-        device1 = _get_vacuum_device_cleaning()
-        component1 = Dyson360EyeDevice(device1)
-        component1.stop()
-        assert device1.pause.called
-
-    def test_set_fan_speed(self):
-        """Test set fan speed vacuum."""
-        device1 = _get_vacuum_device_cleaning()
-        component1 = Dyson360EyeDevice(device1)
-        component1.set_fan_speed("Max")
-        device1.set_power_mode.assert_called_with(PowerMode.MAX)
-
-    def test_start_pause(self):
-        """Test start/pause."""
-        device1 = _get_vacuum_device_charging()
-        component1 = Dyson360EyeDevice(device1)
-        component1.start_pause()
-        assert device1.start.called
-
-        device2 = _get_vacuum_device_pause()
-        component2 = Dyson360EyeDevice(device2)
-        component2.start_pause()
-        assert device2.resume.called
-
-        device3 = _get_vacuum_device_cleaning()
-        component3 = Dyson360EyeDevice(device3)
-        component3.start_pause()
-        assert device3.pause.called
-
-    def test_return_to_base(self):
-        """Test return to base."""
-        device = _get_vacuum_device_pause()
-        component = Dyson360EyeDevice(device)
-        component.return_to_base()
-        assert device.abort.called
+    device.state.state = Dyson360EyeMode.FULL_CLEAN_PAUSED
+    state = await async_update_entity(hass, ENTITY_ID)
+    assert state.state == STATE_OFF
+    assert state.attributes[ATTR_STATUS] == "Paused"
+
+
+async def test_commands(hass: HomeAssistant) -> None:
+    """Test sending commands to the vacuum."""
+    device = await async_setup_dyson(hass, _get_vacuum_device)
+
+    async def _async_call_service(
```

Comment:
Please don't create a helper like this. Just call the services inline.

Reply:
Without this helper function, there is going to be a bunch of redundant code. It does not only call the service. It also handles state update, assertion, and resetting the mock. Maybe I should rename this to something like `_test_command`?

Reply:
There's too much logic in the helper. Please just inline. It's ok with some repetition. Another way would be pytest parametrization to avoid repetition.

Reply:
I rewrite the command test using pytest parametrizing. I think this version is better.